charms/knative-operator/src/charm.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import traceback
+
+from ops.main import main
+from ops.pebble import Layer
+from ops.charm import CharmBase
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus, StatusBase
+from jinja2 import Environment, FileSystemLoader
+from lightkube import ApiError, Client, codecs
+from serialized_data_interface import NoCompatibleVersions, NoVersionsListed, get_interfaces
```

Comment:
Related to above `metadata.yaml` comment, I think we can remove the SDI, relation handlers, etc., at least to start